### DISCUSSION TITLE
How do I tell whether your project is referring to minGW or mingw-w64

### DISCUSSION THREAD

**Discussion by SlySven** [microsoft/vcpkg#18704] [Q&A]
I recently got into a heated discussion with another member of the project I code for who is setting up what is labelled as using a `x64-mingw-dynamic` triplet as a GitHub Action. As the people over at [Software in the Public Interest Inc.](http://mingw.org/Welcome_to_MinGW_org) strongly insist:
> MinGW is a registered trademark of Software in the Public Interest Inc., registration number 86017856; it has been registered on behalf of MinGW.org, and its use by any other project is unauthorized.

I am also aware that their software (which includes `MSYS`) only targets the 32 Bit Windows OS but can run in 64 Bit ones.

However it has finally become clear to me that you (like myself 😁) are using the more modern, up to date and capable of targetting both 32 bit AND 64 bit Windows OS natively, fork that is Mingw-w64 and MSYS2 - homepage: http://mingw-w64.org/doku.php.

Is there anything that can be done to make this important distinction more clear?

**Comment by PhoebeHui** [comment]
I have no idea about this, @strega-nil, could you please take a look?

**Comment by strega-nil** [comment]
This is really weird. I don't know anything about this... cc @ras0219-msft

**Comment by SlySven** [comment]
If it helps to clarify things - you seem to be using the *modern*/*up-to-date* fork that is Mingw-w64 [Wikipedia article](https://en.wikipedia.org/wiki/Mingw-w64) (which targets both 32 and 64 bit Windows directly) rather than the original MinGW [Wikipedia article](https://en.wikipedia.org/wiki/MinGW) that only targets 32-bit Windows but which can run on 64-bit Windows because the latter has Windows-On-Windows64 [Wikipedia article](https://en.wikipedia.org/wiki/WoW64).

Putting it simply, the ambiguity of your naming scheme(s) is confusing for those who know (and care) about the difference...!

*\<aside>It is my opinion that the *original* project does not like the *forked* one, probably (again in my opinion) as the latter is more popular - especially with those who want to make the most of 64-bit processors!\</aside>* 😜

**Comment by longnguyen2004** [comment]
@SlySven If there's a way to differentiate between the 2 forks purely by checking compilers output, then I'll happy to add that in. For now, vcpkg doesn't care about any of that. Whatever version on the PATH will be used. We however use the prefixed name to look for the compiler. So if you use x64-mingw-dynamic, vcpkg will look for `x86_64-w64-mingw32-gcc.exe`, and will happily error out if it doesn't exist.

But personally, I find it a bit silly that there are still people using the old MinGW fork, especially since Mingw-w64 is superior in almost every way.
